lib/spack/spack/fetch_strategy.py

```diff
@@ -1129,15 +1130,18 @@ def fetch(self):
 
         parsed_url = url_util.parse(self.url)
         if parsed_url.scheme != 's3':
-            raise ValueError(
+            raise FetchError(
```

Comment:
(question) Was this causing problems before? I would assume `fetch_strategy.from_url_scheme` would only return an `S3FetchStrategy` if this check would pass. In other words, it would be unexpected to run into this error; or am I wrong about that?

Reply:
tl;dr Trying to use an S3FetchStrategy on a non-s3 URL should still fall-through.  The fact that we never even try is moot, imo.

Yes, you are correct in your observation.  My reasoning with this change is that the check is done at `fetch()` time, and so anything that goes wrong should be some sort of `FetchError()`.

Another point to consider: with this change, we could hypothetically do away with `from_url_scheme()`, and just try `S3FetchStrategy` with every URL.  While I'm not necessarily suggesting we do this, my point is that, if we were to do something like this, this change would be necessary to preserve the fall-through behavior at the heart of #13537.

Reply:
Sounds good - thanks for the explanation!